### PR TITLE
Fix lockup issues found by @ailisp

### PR DESCRIPTION
**Pull request by telezhnaya**
The first commit solves #145 
The second commit adds the instruction on how to build the contract.

All the other commits are addressing issues in [the report](https://hackmd.io/@ailisp/rJbiDYr5_) file-by-file, you could check them separately if you want to.

I also partially fixed issues from https://github.com/near/core-contracts/issues/146, some of them were intersected with the issues from the report.

---

What **haven't** been done during this PR:
- I didn't fix the comments in README. You could see some fixes there because they are required to have a consistent state with the code, but I will re-check the comments from @ailisp separately
- I didn't fix https://github.com/near/core-contracts/issues/139 because I don't feel confident in it. @ailisp does not find the problem there at all. I feel I need more time to dig into that, not sure the issue worth it.
- I didn't mark as deprecated `transfers_information`, `voting`, and other related stuff. While I was rewriting README on the previous iteration, @evgenykuzyakov pointed out that it's not correct to say that transfers were enabled on Oct 13, 2020, because the contract could be invoked not only in the Mainnet, and that date could differ. So, from that point of view, the mechanism should not also be deprecated, it still could be used.
- I didn't rename `foundation_account_id` to `terminator_account_id` because it changes the public interface, so it could break the backward compatibility.

----
@AngelBlock you could be also interested in it

### CODE REVIEW

File: lockup/src/lib.rs [hunk 1]
```diff
@@ -315,8 +313,8 @@ mod tests {
                     vesting_schedule,
                     salt: SALT.to_vec().into(),
                 }
-                .hash()
-                .into(),
+                    .hash()
```
Comment (ailisp) [inline lockup/src/lib.rs:316]: I'm not sure, is this formatting right?

**Review by ailisp — APPROVED** [review]
Lgtm! Nice job

File: lockup/src/lib.rs (hunk 1 above)
Comment (telezhnaya) [reply lockup/src/lib.rs:316]: CLion fixes it that way, and I've seen other places with the same formatting, so I decided to have these changes

**Comment by telezhnaya** [comment]
Rebased because we merged required PR (nothing is changed in this PR)
Now diff is exactly as it should be

**Review by evgenykuzyakov — APPROVED** [review]

**Pull request merged**